ci-operator/step-registry/gcp/provision/dns-peering-zone/sa/gcp-provision-dns-peering-zone-sa-commands.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+python3 --version 
+export CLOUDSDK_PYTHON=python3
+
+function backoff() {
+  local attempt=0
+  local failed=0
+  echo "INFO: Running Command '$*'"
+  while true; do
+    eval "$@" && failed=0 || failed=1
+    if [[ $failed -eq 0 ]]; then
+      break
+    fi
+    attempt=$(( attempt + 1 ))
+    if [[ $attempt -gt 5 ]]; then
+      break
+    fi
+    echo "command failed, retrying in $(( 2 ** attempt )) seconds"
+    sleep $(( 2 ** attempt ))
+  done
+  return $failed
+}
+
+function run_command() {
+    local CMD="$1"
+    echo "Running command: ${CMD}"
+    eval "${CMD}"
+}
+
+ret=0
+
+GOOGLE_PROJECT_ID="$(< ${CLUSTER_PROFILE_DIR}/openshift_gcp_project)"
+export GCP_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/gce.json"
+sa_email=$(jq -r .client_email ${GCP_SHARED_CREDENTIALS_FILE})
+if ! gcloud auth list | grep -E "\*\s+${sa_email}"
+then
+  gcloud auth activate-service-account --key-file="${GCP_SHARED_CREDENTIALS_FILE}"
+  gcloud config set project "${GOOGLE_PROJECT_ID}"
+fi
+
+echo "$(date -u --rfc-3339=seconds) - INFO: reading variables from 'xpn_project_setting.json'..."
```

Comment:
The case need XPN project ?

Reply:
No. We take the XPN project as the target project of the DNS peering zone, so what it needs here is the project id and the network name in the project.